test/helper/common.bash

```diff
@@ -117,131 +232,207 @@ function repeat_in_container_until_success_or_timeout() {
-    # https://github.com/docker-mailserver/docker-mailserver/pull/2997#discussion_r1070583876
-    docker exec "${CONTAINER_NAME}" grep --count "${MATCH_CONTENT}" "${MATCH_IN_LOG}" || true
+    _exec_in_container grep --count "${MATCH_CONTENT}" "${MATCH_IN_LOG}"
   }
 
   # WARNING: Keep in mind it is a '>=' comparison.
   # If you provide an explict count to match, ensure it is not too low to cause a false-positive.
   function __has_expected_count() {
+    # shellcheck disable=2317
     [[ $(__get_count) -ge "${EXPECTED_COUNT}" ]]
   }
 
-  local CONTAINER_NAME=${1}
-  local EXPECTED_COUNT=${2}
-
-  # Ensure early failure if arg is missing:
-  assert_not_equal "${CONTAINER_NAME}" ''
+  local CONTAINER_NAME=${1:?Container name must be provided}
+  local EXPECTED_COUNT=${2:-}
 
   # Ensure the container is configured with the required `LOG_LEVEL` ENV:
   assert_regex \
-    $(docker exec "${CONTAINER_NAME}" env | grep '^LOG_LEVEL=') \
+    "$(_exec_in_container env | grep '^LOG_LEVEL=')" \
     '=(debug|trace)$'
 
   # Default behaviour is to wait until one new match is found (eg: incremented),
   # unless explicitly set (useful for waiting on a min count to be reached):
-  if [[ -z $EXPECTED_COUNT ]]
-  then
-    # +1 of starting count:
-    EXPECTED_COUNT=$(( $(__get_count) + 1 ))
-  fi
+  #
+  #                             +1 of starting count:
+  [[ -z ${EXPECTED_COUNT} ]] && EXPECTED_COUNT=$(bc <<< "$(__get_count) + 1")
 
   repeat_until_success_or_timeout 20 __has_expected_count
 }
 
 # An account added to `postfix-accounts.cf` must wait for the `changedetector` service
-# to process the update before Dovecot creates the mail account and associated storage dir:
-function wait_until_account_maildir_exists() {
-  local CONTAINER_NAME=$1
-  local MAIL_ACCOUNT=$2
+# to process the update before Dovecot creates the mail account and associated storage dir.
+#
+# @param ${1} = container name
+function _wait_until_account_maildir_exists() {
+  local CONTAINER_NAME=${1:?Container name must be provided}
+  local MAIL_ACCOUNT=${2:?Mail account must be provided}
 
   local LOCAL_PART="${MAIL_ACCOUNT%@*}"
   local DOMAIN_PART="${MAIL_ACCOUNT#*@}"
   local MAIL_ACCOUNT_STORAGE_DIR="/var/mail/${DOMAIN_PART}/${LOCAL_PART}"
 
-  repeat_in_container_until_success_or_timeout 60 "${CONTAINER_NAME}" bash -c "[[ -d ${MAIL_ACCOUNT_STORAGE_DIR} ]]"
+  _repeat_in_container_until_success_or_timeout 60 "${CONTAINER_NAME}" \
+    /bin/bash -c "[[ -d ${MAIL_ACCOUNT_STORAGE_DIR} ]]"
+}
+
+# Wait until the mail queue is empty inside a container (${1}).
+#
+# @param ${1} = container name
+function _wait_for_empty_mail_queue_in_container() {
+  local CONTAINER_NAME=${1:?Container name must be provided}
+  local TIMEOUT=${TEST_TIMEOUT_IN_SECONDS}
+
+  # shellcheck disable=SC2016
+  _repeat_in_container_until_success_or_timeout \
+    "${TIMEOUT}" \
+    "${CONTAINER_NAME}" \
+    /bin/bash -c '[[ $(mailq) == *"Mail queue is empty"* ]]'
```

Comment:
Being more explicit should be ok here. From what I've seen it's the only output from the command when valid.

```suggestion
    /bin/bash -c '[[ $(mailq) == "Mail queue is empty" ]]'
```

Reply:
Done